src/core/thread/router_table.cpp

```diff
@@ -875,8 +864,17 @@ void RouterTable::RouterIdMap::HandleTimeTick(void)
     }
 }
 
+void RouterTable::SignalTableChanged(void) { mChangedTask.Post(); }
+
+void RouterTable::HandleTableChanged(void)
+{
+#if OT_SHOULD_LOG_AT(OT_LOG_LEVEL_INFO)
+    LogRouteTable();
+#endif
+}
+
 #if OT_SHOULD_LOG_AT(OT_LOG_LEVEL_INFO)
-void RouterTable::LogRouteTable(void) const
+void RouterTable::LogRouteTable(void)
```

Comment:
Can we keep the `const` declaration here?

Reply:
This is the change to work around the `clang-tidy` warning/error (above comment)

If we keep it as `const` then, `clang-tidy` will want the `HandleTableChanged()` to be defined as `const` as well (which we cannot do since we use it as callback in `TaskletIn` which expects a non-const method pointer). I guess `clang-tidy` cannot realize this aspect of the use of `HandleTableChanged()` method (as a method pointer).

Reply:
That makes sense.

What was the part about "If we use any private variable, clang-tidy will not receommend it anymore." above?

Reply:
Basically `clang-tidy` does not add warning/error that `LogRouteTable()` should be `const` (which it can be technically) because we access/read private variable (to me it seems like kind of a weird rule, probably to address some edge-case, but we can take advantage of it here).

BTW, I will later add callback from `HandleTableChanged()` to `HistoryTracker` to track router table changes. I think with that we may be able to again mark `LogRouteTable()` as `const` while defining the `HandleTableChanged()` as non-const and I think clang-tidy may be ok with it.